### PR TITLE
Global accelerator active active

### PR DESCRIPTION
**Pull request by ryanemerson**
# Create deployment with accelerator

1. Create Accelerator: `ACCELERATOR_NAME= CLUSTER_1= CLUSTER_2= KEYCLOAK_NAMESPACE= ./provision/aws/global-accelerator/accelerator_multi_az_create.sh`
2. Take note of `ACCELERATOR DNS` value from stdout
3. cd `provision/rosa-cross-dc`
4. Deploy Keycloak: `KC_CLIENT_URL=<ACCELERATOR DNS> ROSA_CLUSTER_NAME_1= ROSA_CLUSTER_NAME_2= AURORA_CLUSTER= AURORA_REGION= task`

# Remove accelerator and LBs
1. `ACCELERATOR_NAME= CLUSTER_1= CLUSTER_2= KEYCLOAK_NAMESPACE= ./provision/aws/global-accelerator/accelerator_multi_az_delete.sh`

### CODE REVIEW

**Comment by ryanemerson** [comment]
First benchmark results. There's a number of 503 errors, but this is when the number of active users is well over 1000, so I think this is down to load-shedding as there's no exceptions in any of the server logs.

Command
````
./benchmark.sh ${REGION} \
  --scenario=keycloak.scenario.authentication.AuthorizationCode \
  --server-url=${KC_CLIENT_URL} \
  --users-per-sec=300 \
  --measurement=300 \
  --realm-name=realm-0 \
  --logout-percentage=100 \
  --users-per-realm=10000 \
  --ramp-up=100 \
  --log-http-on-failure
```

Results:
![Screenshot from 2024-01-17 11-03-26](https://github.com/keycloak/keycloak-benchmark/assets/765332/a4cc18e1-4d19-4409-9730-a01703aeb123)

![Screenshot from 2024-01-17 11-03-41](https://github.com/keycloak/keycloak-benchmark/assets/765332/03d824f0-07e1-4934-880b-d8c10be96508)

File: provision/aws/global-accelerator/accelerator_multi_az_create.sh [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+ENDPOINT_GROUP_ARN=$(aws globalaccelerator create-endpoint-group \
+  --listener-arn ${LISTENER_ARN} \
+  --traffic-dial-percentage 100 \
+  --endpoint-configurations ${ENDPOINTS} \
+  --endpoint-group-region ${ENDPOINT_GROUP_REGION}
+)
+
+echo "ACCELERATOR DNS: ${ACCELERATOR_DNS}"
+echo "ACCELERATOR DUAL_STACK DNS: ${ACCELERATOR_DNS}"
```
Comment (pruivo) [inline provision/aws/global-accelerator/accelerator_multi_az_create.sh:127]: Do we need ipv6?
```suggestion
echo "ACCELERATOR DUAL_STACK DNS: ${ACCELERATOR_DUAL_STACK_DNS}"
```

Comment (ryanemerson) [reply provision/aws/global-accelerator/accelerator_multi_az_create.sh:127]: Good question! @ahus1 any thoughts?

Comment (ahus1) [reply provision/aws/global-accelerator/accelerator_multi_az_create.sh]: With IPv4 addresses costing money at AWS AFAIK as well as my local provider, and IPv6 is free, plus the world moving to IPv6, I suppose IPv6 is the thing.

Comment (ryanemerson) [reply provision/aws/global-accelerator/accelerator_multi_az_create.sh]: The configuration options are `IPV4` or `DUAL_STACK` so it's not possible to exclusively use IPv6. I'll leave it as `DUAL_STACK` for now.

**Comment by ryanemerson** [comment]
Second run, this time with no 503s just a single failure caused by a SSL handshake timeout (there are no exceptions in the Keycloak or Infinispan logs):

```
15:42:15.745 [DEBUG] i.g.h.e.r.DefaultStatsProcessor - Request 'Browser to Log In Endpoint' failed for user 50787: i.n.h.s.SslHandshakeTimeoutException: handshake timed out after 10000ms
15:42:15.755 [DEBUG] i.g.h.e.r.DefaultStatsProcessor - 
>>>>>>>>>>>>>>>>>>>>>>>>>>
Request:
Browser to Log In Endpoint: KO i.n.h.s.SslHandshakeTimeoutException: handshake timed out after 10000ms
=========================
Session:
Session(Authentication - Authorization Code Username/Password,50787,Map(clientId -> client-0, email -> , gatling.http.cache.dns -> io.gatling.http.resolver.ShufflingNameResolver@6a4692c7, state -> ff37d7df-9a92-a0d0-21a6-e4f15e839a77, redirectUri -> https://aec644a932990cb6f.awsglobalaccelerator.com/realms/realm-0/account, username -> user-5247, scope -> openid profile, adminUsername -> null, lastName -> , firstName -> , gatling.http.ssl.sslContexts -> io.gatling.http.util.SslContexts@1e7bb813, wrongPasswordCount -> 0, adminPassword -> null, clientSecret -> client-0-secret, realm -> realm-0, keycloakServer -> https://aec644a932990cb6f.awsglobalaccelerator.com, password -> user-5247-password),KO,List(),io.gatling.core.protocol.ProtocolComponentsRegistry$$Lambda$596/0x00007feabc4e2668@2902d262,io.netty.channel.epoll.EpollEventLoop@473b3b7a)
=========================
HTTP request:
GET https://aec644a932990cb6f.awsglobalaccelerator.com/realms/realm-0/protocol/openid-connect/auth?login=true&response_type=code&client_id=client-0&state=ff37d7df-9a92-a0d0-21a6-e4f15e839a77&redirect_uri=https%3A%2F%2Faec644a932990cb6f.awsglobalaccelerator.com%2Frealms%2Frealm-0%2Faccount&scope=openid+profile
headers=
Accept: text/html,application/xhtml+xml,application/xml
Accept-Language: en-US,en;q=0.5
accept: application/json
Accept-Encoding: gzip, deflate
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:16.0) Gecko/20100101 Firefox/16.0
host: aec644a932990cb6f.awsglobalaccelerator.com
=========================
HTTP response:
<<<<<<<<<<<<<<<<<<<<<<<<<
```

Screenshot from 2024-01-17 15-47-58
Screenshot from 2024-01-17 15-48-12
Screenshot from 2024-01-17 15-48-24

File: doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+----
+cat <<EOF | oc apply -n $NAMESPACE -f - #<1>
+  apiVersion: v1
+  kind: Service
+  metadata:
+    name: accelerator-loadbalancer
+    annotations:
+      service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
+      service.beta.kubernetes.io/aws-load-balancer-healthcheck-path: "/lb-check"
+      service.beta.kubernetes.io/aws-load-balancer-healthcheck-protocol: "https"#
```
Comment (pruivo) [inline doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc:53]:
```suggestion
      service.beta.kubernetes.io/aws-load-balancer-healthcheck-protocol: "https"
```

File: doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+Perform the following on each of the Keycloak clusters:
++
+.. Login to the ROSA cluster
++
+.. Create a Kubernetes LoadBalancer service
++
+.Command:
+[source,bash]
+----
+cat <<EOF | oc apply -n $NAMESPACE -f - #<1>
```
Comment (pruivo) [inline doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc:45]: I think we are using mostly `kubectl`
```suggestion
cat <<EOF | kubectl apply -n $NAMESPACE -f - #<1>
```

File: doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+<2> How frequently the healthcheck probe is executed in seconds
+<3> How many healthchecks must pass for the NLB to be considered healthy
+<4> How many healthchecks must fail for the NLB to be considered unhealthy
++
+.. Take note of the DNS hostname as this will be required later:
++
+.Command:
+[source,bash]
+----
+oc -n $NAMESPACE get svc accelerator-loadbalancer --template="{{range .status.loadBalancer.ingress}}{{.hostname}}{{end}}"
```
Comment (pruivo) [inline doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc:81]:
```suggestion
kubectl -n $NAMESPACE get svc accelerator-loadbalancer --template="{{range .status.loadBalancer.ingress}}{{.hostname}}{{end}}"
```

File: doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+[source,json]
+----
+{
+    "EndpointGroup": {
+        "EndpointGroupArn": "arn:aws:globalaccelerator::606671647913:accelerator/e35a94dd-391f-4e3e-9a3d-d5ad22a78c71/listener/1f396d40/endpoint-group/2581af0dc700",
+        "EndpointGroupRegion": "eu-west-1",
+        "EndpointDescriptions": [
+            {
+                "EndpointId": "arn:aws:elasticloadbalancing:eu-west-1:606671647913:loadbalancer/net/abab80a363ce8479ea9c4349d116bce2/6b65e8b4272fa4b5",
+                "Weight": 100,
```
Comment (pruivo) [inline doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc:222]: is this correct or should be 50?

File: doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+                "HealthState": "HEALTHY",
+                "ClientIPPreservationEnabled": false
+            },
+            {
+                "EndpointId": "arn:aws:elasticloadbalancing:eu-west-1:606671647913:loadbalancer/net/a1c76566e3c334e4ab7b762d9f8dcbcf/985941f9c8d108d4",
+                "Weight": 0,
+                "HealthState": "UNHEALTHY",
```
Comment (pruivo) [inline doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc:229]: `UNHEALTHY`?

**Review by pruivo — CHANGES_REQUESTED** [review]
Looks good to me, just minor suggestions :+1:

File: doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc (hunk 5 above)
Comment (ryanemerson) [reply doc/kubernetes/modules/ROOT/pages/running/loadbalancing.adoc:222]: It should be 50 :+1:

**Review by pruivo — APPROVED** [review]

**Pull request merged**